packages/core/src/services/workspaces/usage.ts

```diff
@@ -0,0 +1,61 @@
+import { count, gte } from 'drizzle-orm'
+
+import { MAX_FREE_RUNS, Workspace, WorkspaceUsage } from '../../browser'
+import { database } from '../../client'
+import { PromisedResult, Result } from '../../lib'
+import { EvaluationResultsRepository } from '../../repositories'
+import { DocumentLogsRepository } from '../../repositories/documentLogsRepository'
+
+export function getLatestRenewalDate(firstRenewalDate: Date, targetDate: Date) {
+  if (targetDate.getTime() < firstRenewalDate.getTime()) {
+    return firstRenewalDate
+  }
+
+  const day = firstRenewalDate.getDate() // The day of the month is always mantained.
```

Comment:
careful with utc shenanigans if any of your dates is not in utc you could have unexpected variations